master/internal/project/authz_basic_impl.go

```diff
@@ -0,0 +1,121 @@
+package project
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/projectv1"
+	"github.com/determined-ai/determined/proto/pkg/workspacev1"
+)
+
+// ProjectAuthZBasic is classic OSS Determined authentication for projects.
+type ProjectAuthZBasic struct{}
+
+// CanGetProject always return true and a nil error for basic auth.
+func (a *ProjectAuthZBasic) CanGetProject(
+	curUser model.User, project *projectv1.Project,
+) (canGetProject bool, serverError error) {
+	return true, nil
```

Comment:
why does this return a pair of bool and error, and not just an error like the methods after?

Reply:
The pair is to allow us to prevent information leakage (the existence of a resource). The error is for some db error that should be bubbled up to the user and the boolean is to indicate that the user should not be able to view the resource. 

Whether this should be the case, ties into the question of system clarity vs security which I will raise a discussion about.